enhydris_api_client/__init__.py

```diff
@@ -22,95 +16,94 @@ def urljoin(*args):
-        cookies=r.cookies,
-        allow_redirects=False,
-    )
-    r1.raise_for_status()
-    result = r1.cookies
-    return result
-
-
-def get_model(base_url, session_cookies, model, id):
-    url = urljoin(base_url, "api/{}/{}/".format(model, id))
-    r = requests.get(url, cookies=session_cookies)
-    r.raise_for_status()
-    return r.json()
-
-
-def post_model(base_url, session_cookies, model, data):
-    r = requests.post(
-        urljoin(base_url, "api/{}/".format(model)),
-        headers={
-            "X-CSRFToken": session_cookies["csrftoken"],
-            "Content-Type": "application/x-www-form-urlencoded"
-            # See comment about x-www-form-urlencoded above
-        },
-        cookies=session_cookies,
-        data=data,
-    )
-    r.raise_for_status()
-    return r.json()["id"]
-
-
-def delete_model(base_url, session_cookies, model, id):
-    url = urljoin(base_url, "api/{}/{}/".format(model, id))
-    r = requests.delete(
-        url,
-        cookies=session_cookies,
-        headers={"X-CSRFToken": session_cookies["csrftoken"]},
-    )
-    if r.status_code != 204:
-        raise requests.exceptions.HTTPError()
-
-
-def read_tsdata(base_url, session_cookies, ts_id):
-    url = urljoin(base_url, "api/tsdata/{0}/".format(ts_id))
-    r = requests.get(url, cookies=session_cookies)
-    r.raise_for_status()
-    return pd.read_csv(StringIO(r.text), header=None, parse_dates=True, index_col=0)
-
-
-def post_tsdata(base_url, session_cookies, timeseries_id, ts):
-    f = StringIO()
-    ts.to_csv(f, header=False)
-    url = urljoin(base_url, "api/tsdata/{}/".format(timeseries_id))
-    r = requests.post(
-        url,
-        data={"timeseries_records": f.getvalue()},
-        headers={
-            "Content-Type": "application/x-www-form-urlencoded",
-            "X-CSRFToken": session_cookies["csrftoken"],
-        },
-        cookies=session_cookies,
-    )
-    r.raise_for_status()
-    return r.text
-
-
-def get_ts_end_date(base_url, session_cookies, ts_id):
-    url = urljoin(base_url, "timeseries/d/{}/bottom/".format(ts_id))
-    r = requests.get(url, cookies=session_cookies)
-    r.raise_for_status()
-    lines = r.text.splitlines()
-    lines.reverse()
-    for line in [x.strip() for x in lines]:
-        if not line:
-            continue
-        datestring = line.split(",")[0]
-        return iso8601.parse_date(datestring, default_timezone=None)
-    return datetime(1, 1, 1)
+class EnhydrisApiClient:
+    def __init__(self, base_url):
+        self.base_url = base_url
+        self.cookies = {}
+
+    def login(self, username, password):
+        self.cookies = {}
+        if not username:
+            return
+        login_url = urljoin(self.base_url, "accounts/login/")
+        r = requests.get(login_url)
+        r.raise_for_status()
+        r1 = requests.post(
+            login_url,
+            headers={
+                "X-CSRFToken": r.cookies.get("csrftoken", "unspecified CSRF token"),
```

Comment:
If "X-CSRTOKEN" can take the default value of "unspecified CSRF token" why we are trying to get it?